packages/core/src/api/exporters/html/htmlConversion.test.ts

```diff
@@ -107,144 +116,304 @@ describe("Test HTML conversion", () => {
+      //   {
+      //     type: "table",
+      //     content: {
+      //       type: "tableContent",
+      //       rows: [
+      //         {
+      //           cells: ["Table Cell", "Table Cell"],
+      //         },
+      //         {
+      //           cells: ["Table Cell", "Table Cell"],
+      //         },
+      //       ],
+      //     },
+      //   },
+      // ],
+    },
+  ];
+
+  let pmView: any;
   let editor: BlockNoteEditor;
   const div = document.createElement("div");
+  document.body.append(div);
+
   beforeEach(() => {
+    editor.replaceBlocks(editor.document, initialContent);
+  });
+
+  beforeAll(async () => {
+    pmView = await import("prosemirror-view");
+    (window as any).__TEST_OPTIONS = (window as any).__TEST_OPTIONS || {};
+
     editor = BlockNoteEditor.create();
     editor.mount(div);
+
+    await initializeESMDependencies();
   });
 
-  afterEach(() => {
+  afterAll(() => {
     editor.mount(undefined);
     editor._tiptapEditor.destroy();
     editor = undefined as any;
 
     delete (window as Window & { __TEST_OPTIONS?: any }).__TEST_OPTIONS;
   });
 
-  // When the selection starts in a nested block, the Fragment from it omits the
-  // `blockContent` node of the parent `blockContainer` if it's not also
-  // included in the selection. In the schema, `blockContainer` nodes should
-  // contain a single `blockContent` node, so this edge case needs to be tested.
-  describe("No block content", () => {
-    const blocks: PartialBlock[] = [
-      {
-        type: "paragraph",
-        content: "Paragraph 1",
-        children: [
-          {
-            type: "paragraph",
-            content: "Nested Paragraph 1",
-          },
-          {
-            type: "paragraph",
-            content: "Nested Paragraph 2",
-          },
-          {
-            type: "paragraph",
-            content: "Nested Paragraph 3",
-          },
-        ],
-      },
-      {
-        type: "paragraph",
-        content: "Paragraph 2",
-        children: [
-          {
-            type: "paragraph",
-            content: "Nested Paragraph 1",
-          },
-          {
-            type: "paragraph",
-            content: "Nested Paragraph 2",
-          },
-          {
-            type: "paragraph",
-            content: "Nested Paragraph 3",
-          },
-        ],
-      },
-    ];
-
-    beforeEach(() => {
-      editor.replaceBlocks(editor.document, blocks);
-    });
-
-    it("Selection within a block's children", async () => {
-      // Selection starts and ends within the first block's children.
+  // Sets the editor selection to the given start and end positions, then
+  // exports the selected content to HTML and compares it to a snapshot.
+  function testSelection(testName: string, startPos: number, endPos: number) {
+    editor.dispatch(
+      editor._tiptapEditor.state.tr.setSelection(
+        TextSelection.create(editor._tiptapEditor.state.doc, startPos, endPos)
+      )
+    );
+    if (
```

Comment:
does this actually happen? Because you `setSelection` to a `TextSelection`, not a `NodeSelection` which I think doesn't have a node property.